### PR TITLE
WT-10084 Minor refactor of cursor search near.

### PR DESCRIPTION
**Pull request by luke-pearson**
…he top level function.

### CODE REVIEW

**Comment by luke-pearson** [comment]
@agorrod @keitharnoldsmith this is an offshoot of my WT-9901 PR. Which I'm not sure about merging yet, I wanted to keep the refactor changes though and merge them separately that way we can still have some of the benefit.

File: src/btree/bt_cursor.c [hunk 1]
```diff
@@ -888,6 +835,104 @@ __wt_btcur_search(WT_CURSOR_BTREE *cbt)
     return (ret);
 }
 
+/*
+ * __btcur_search_neighboring --
+ *     Try after the search key, then before. At low isolation levels, new records could appear as
+ *     we are stepping through the tree.
+ */
+static int
+__btcur_search_neighboring(WT_CURSOR_BTREE *cbt, WT_CURFILE_STATE *state, int *exact)
```
Comment (luke-pearson) [inline src/btree/bt_cursor.c:844]: I renamed and moved this closer to the caller so there's less code walking when reading this code.

**Review by keitharnoldsmith — APPROVED** [review]
LGTM!

**Comment by keitharnoldsmith** [comment]
Just curious about the name change from `__cursor_search_neighboring()` to `__btcur_search_neighboring()`. The internal functions in `bt_cursor.c` seem to have an even mix of `__cursor...` and `__btcur...` names. Is there some internal logic to where we prefer one over the other?

**Comment by luke-pearson** [comment]
> Just curious about the name change from `__cursor_search_neighboring()` to `__btcur_search_neighboring()`. The internal functions in `bt_cursor.c` seem to have an even mix of `__cursor...` and `__btcur...` names. Is there some internal logic to where we prefer one over the other?

I would agree that it is confusing, I was under the impression that generally things in the bt_cursor file should start with btcur.

**Review by agorrod — APPROVED** [review]
LGTM - thanks Luke, this is a nice improvement.

**Pull request merged**